### ISSUE TITLE
poetry install fails:  dyld: Library not loaded: @executable_path/../Python3 on macOS 11.6

### ISSUE THREAD

**Issue by jlixfeld** [python-poetry/poetry#5109]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
-- N/A - cannot install poetry
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).
-- N/A - cannot install poetry

- **OS version and name**: macOS 11.6
- **Poetry version**: N/A
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: N/A

## Issue

Poetry install fails:

```
jlixfeld@Jasons-MacBook-Pro ~ % curl -sSL https://install.python-poetry.org | python3 -
Retrieving Poetry metadata

# Welcome to Poetry!

This will download and install the latest version of Poetry,
a dependency and package manager for Python.

It will add the `poetry` command to Poetry's bin directory, located at:

/Users/jlixfeld/Library/Python/3.8/bin

You can uninstall at any time by executing this script with the --uninstall option,
and these changes will be reverted.

Installing Poetry (1.1.12): An error occurred. Removing partial environment.
Poetry installation failed.
See /Users/jlixfeld/poetry-installer-error-h3v75uw9.log for error logs.
jlixfeld@Jasons-MacBook-Pro ~ % more /Users/jlixfeld/poetry-installer-error-h3v75uw9.log
dyld: Library not loaded: @executable_path/../Python3
  Referenced from: /Users/jlixfeld/Library/Application Support/pypoetry/venv/bin/python3
  Reason: image not found

Traceback:

  File "<stdin>", line 877, in main
  File "<stdin>", line 514, in run
jlixfeld@Jasons-MacBook-Pro ~ % 
```

There is no previous poetry install or reference to poetry in `caches` or `Application Support`:

```
jlixfeld@Jasons-MacBook-Pro ~ % ls -al Library/Application\ Support/pypoetry
total 0
drwxr-xr-x   2 jlixfeld  staff    64 26 Jan 16:58 .
drwx------+ 55 jlixfeld  staff  1760 26 Jan 16:43 ..
jlixfeld@Jasons-MacBook-Pro ~ % ls -al Library/Caches/pypoetry
ls: Library/Caches/pypoetry: No such file or directory
```

Python version 3.8.9:
```
jlixfeld@Jasons-MacBook-Pro ~ % python3 --version
Python 3.8.9
jlixfeld@Jasons-MacBook-Pro ~ %
```

Xcode version:
```
jlixfeld@Jasons-MacBook-Pro ~ % xcodebuild -version
Xcode 13.2.1
Build version 13C100
jlixfeld@Jasons-MacBook-Pro ~ %
```

**Comment by didac-94** [comment]
Just had the same issue on MacOS 12.2, also no previous installations of poetry, Python 3.8.9 but xcode was not installed (only CLI tools).

Solved it by installing python through homebrew (though it was 3.9.1). Installation process went fine then.

**Comment by jlixfeld** [comment]
Indeed.  I came to more or less the same resolution.  I've since re-installed the system with macOS 12.2, then through `homebrew`, I installed `pyenv`, used it to install `3.8.12`, then set `pyenv local 3.8.12` and installed `poetry`.  At that point the installation went fine.

**Comment by jacob-lee** [comment]
mac)S 11.6.3

Same issue. Only system python installed. Installed python from macports, which seemed to resolve that issue (though another came up...).

**Comment by amabri04** [comment]
Similar issue on my end but I cannot get it working :(

I installed python (3.9) using `homebrew` prior to installing `Poetry`. I tried re-installing python again but it does not fix the issue.

Here the output when trying to install Poetry:
```
amabri@amabris-MacBook-Pro ~ % curl -sSL https://install.python-poetry.org | POETRY_VERSION=1.1.12 python3 - 
Retrieving Poetry metadata

# Welcome to Poetry!

This will download and install the latest version of Poetry,
a dependency and package manager for Python.

It will add the `poetry` command to Poetry's bin directory, located at:

/Users/amabri/Library/Python/3.8/bin

You can uninstall at any time by executing this script with the --uninstall option,
and these changes will be reverted.

Installing Poetry (1.1.12): An error occurred. Removing partial environment.
Poetry installation failed.
See /Users/amabri/poetry-installer-error-gznvefo1.log for error logs.
```

The error log is showing the following:
```
dyld[24372]: Library not loaded: @executable_path/../Python3
  Referenced from: /Users/amabri/Library/Application Support/pypoetry/venv/bin/python3
  Reason: tried: '/Users/amabri/Library/Application Support/pypoetry/venv/bin/../Python3' (no such file), '/usr/local/lib/Python3' (no such file), '/usr/lib/Python3' (no such file)

Traceback:

  File "<stdin>", line 879, in main
  File "<stdin>", line 517, in run
```

I did a bit of debug and I do not know if the issue could come from here but I might have a weird behaviour with python3 version which does not match the version installed with homebrew.

```
amabri@amabris-MacBook-Pro bin % python3 --version
Python 3.8.9
```

But the symlink seems correct:
```
amabri@amabris-MacBook-Pro bin % ls -l | grep python3
lrwxr-xr-x  1 amabri  admin       39 21 Feb 12:44 python3 -> ../Cellar/python@3.9/3.9.10/bin/python3
```

**Comment by jar349** [comment]
Although it seems pretty well established at this point, I too confirm this problem with system-install Python on Monterey, Intel chip mac. Installed pyenv and 3.9.11 and re-ran the poetry install with great success.

**Comment by Ruslanmsv** [comment]
Same here. I've got new M1 pro macbook pro 14" with macos Monterey and preinstalled python 3.8.9. 
Trying to install poetry with [official installer](https://python-poetry.org/docs/master/#installing-with-the-official-installer) resulted in error

```
dyld[16230]: Library not loaded: @executable_path/../Python3
  Referenced from: /Users/ruslanmsv/Library/Application Support/pypoetry/venv/bin/python3
  Reason: tried: '/Users/ruslanmsv/Library/Application Support/pypoetry/venv/bin/../Python3' (no such file), '/usr/local/lib/Python3' (no such file), '/usr/lib/Python3' (no such file)

Traceback:

  File "<stdin>", line 879, in main
  File "<stdin>", line 517, in run
```

**Comment by ccazabon** [comment]
This still happens (M1 Mac, macOS 12.3.1 Monterey).

It would be nice if poetry's installer could at least detect this situation - running on Mac, python interpreter is the base OS installed one) and give a warning / link / whatever.

**Comment by madisonbullard** [comment]
I fixed this issue by adding `/Users/{your_username_here}/.poetry/bin` to my `PATH` in `.zshrc`. The Poetry installer says it does this for me, but in my case it did not. Once I added the path manually and restarted my terminal, the error was gone 👍🏻

**Comment by mikhail-tokarev** [comment]
This is still reproducible on M1, macOS 12.4 Monterey if you use `curl -sSL https://install.python-poetry.org | python3 -` script. 

However worked well when I tried 
```
curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python 
```

**Comment by mionker** [comment]
The install works if the virtual environment is created using symlinks instead of copying the macOS system python3 executable into the venv. The macOS system python3 will apparently not run outside its standard location - at least not the actual python3 executable in 

`/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/bin/python3.8`

The python3 in 

`/usr/bin/python3`

appears to be a shim to the actual python3.8 executable.

Anyways I got the standard poetry install script to work when I changed the following line

`builder = venv.EnvBuilder(clear=True, with_pip=True, symlinks=False)`

in the VirtualEnvironment make method to

`builder = venv.EnvBuilder(clear=True, with_pip=True, symlinks=True)`

I cannot comment if this ia otherwise a good idea or whether we should simply use a separately installed python instead.
